### DISCUSSION TITLE
How to upload HLS file to GCS specific folder?

### DISCUSSION THREAD

**Discussion by LeoHsu0705** [ant-media/Ant-Media-Server#6406] [FAQs]
I want to achieve this effect.
is there any way to set the stream recording?

```
streams
├─ folder1
│  ├─ 2024-06-16
│  │  ├─ video.m3u8
│  │  ├─ video0001.ts
│  │  └─ video0002.ts
│  ├─ 2024-06-17
│  │  ├─ video.m3u8
│  │  ├─ video0001.ts
│  │  └─ video0002.ts
├─ folder2
│  ├─ 2024-06-16
│  │  ├─ video.m3u8
│  │  ├─ video0001.ts
│  │  └─ video0002.ts
│  ├─ 2024-06-17
│  │  ├─ video.m3u8
│  │  ├─ video0001.ts
│  │  └─ video0002.ts
└─ folder3
```

**Comment by yashtandon113** [comment]
Hi @LeoHsu0705 
You can use the **subFolder** field while creating the stream to achieve the required functionality of having HLS files into the dates subfolder. Please check the below document for references.
https://antmedia.io/docs/guides/playing-live-stream/hls-playing/#playing-streams-from-subfolders

The same subfolders will be created on the bucket side as well when the files are uploaded.

I hope it helps.

**Reply by LeoHsu0705** [reply]
OK,thanks for answer! I will try it.

**Reply by LeoHsu0705** [reply]
I tried.When the stream is published without an API call (eg. a direct publishing over RTMP), there's no subfolder and all files are stored directly in {AppsName}/streams/

**Reply by yashtandon113** [reply]
@LeoHsu0705 
I think we missed your last comment.

Yes, it is expected because currently subFolder can only be used while creating the stream via API.

Can you please try adding the folder name to the RTMP end point as below?

**Endpoint:-** rtmp://domain/LiveApp/**subFoldername**
**Streamkey:-** your-stream-Id

In this way, the subfolder will be created inside the stream folder. It works on the server side but I am not sure if that will work for s3 upload as well or not.

Please let us know if that helps.
